benchmarks/benchmarks/stats.py

```diff
@@ -174,3 +178,51 @@ def time_binned_statistic_dd_reuse_bin(self, statistic):
         stats.binned_statistic_dd(
             [self.inp[0], self.inp[1]], self.inp[2], statistic=statistic,
             binned_statistic_result=self.ret)
+
+
+class ContinuousFitAnalyticalMLEOverride(Benchmark):
+    # list of distributions to time
+    dists = ["pareto", "laplace", "beta", "ncf"]
+    # add custom values for rvs and fit, if desired, for any distribution:
+    # key should match name in dists and value should be list of loc, scale,
+    # and shapes
+    custom_input = {"beta": [1, 1, 1, 1]}
+    fnames = ['floc', 'fscale', 'f0', 'f1', 'f2']
+    fixed = {}
+
+    param_names = ["distribution", "loc_fixed", "scale_fixed",
+                   "shape1_fixed", "shape2_fixed", "shape3_fixed"]
+    params = [dists, * [[True, False]] * 5]
+
+    def setup(self, dist_name, loc_fixed, scale_fixed, shape1_fixed,
+              shape2_fixed, shape3_fixed):
+        exec("self.distn= stats." + dist_name)
+
+        nparam = (len(self.distn.shapes.replace(',', ' ').split())
+                  if self.distn.shapes else 0) + 2
+        all_parameters = [loc_fixed, scale_fixed, shape1_fixed, shape2_fixed,
+                          shape3_fixed]
+        relevant_parameters = all_parameters[:nparam]
+        nonrelevant_parameters = all_parameters[nparam:]
+
+        # skip if all parameters are fixed or if non relevant parameters are
+        # not all false
+        if True in nonrelevant_parameters or False not in relevant_parameters:
+            raise NotImplementedError("skip non-relevant case")
+
+        # use custom input or default loc, scale, and shape.
+        # default shapes are from `_distr_params.py`
+        default_shapes = [i for i in distcont if i[0] == dist_name][0][1]
```

Comment:
I thought there had to be another way, but I didn't know what it was. Believe it or not:
```suggestion
        default_shapes = dict(distcont)[dist_name]
```

Reply:
As I was writing that list comprehension I cringed, thinking there had to be a better way. 

Do you think that running `distcont = dict(distcont)` as an attribute of the class rather than inside the setup method would be better?

Then in each setup method it would be
`default_shapes  = distcont[dist_name]`

Reply:
There would be a difference if there were multiple instances of the class or if setup ran more than once, but I don't think those apply. I think either is fine.